Extensions/disable_gifs.js

```diff
@@ -36,6 +63,42 @@ XKit.extensions.disable_gifs = new Object({
 
 	},
 
+	react_do: function() {
+		const {react_draw} = XKit.extensions.disable_gifs;
+
+		$('figure img[srcset*=".gif"]:not(.xkit-disabled-gif)').each(function() {
+			$(this).addClass('xkit-disabled-gif');
+
+			if (this.complete && this.currentSrc) {
+				react_draw(this);
+			} else {
+				this.onload = function() { react_draw(this); };
+			}
+		});
+	},
+
+	react_draw: function(gif) {
+		const image = new Image();
+		image.src = gif.currentSrc;
+		image.onload = () => {
+			const canvas = document.createElement('canvas');
+			canvas.width = image.naturalWidth;
+			canvas.height = image.naturalHeight;
+			canvas.className = gif.className;
+			canvas.classList.remove('xkit-disabled-gif');
+			canvas.classList.add('xkit-paused-gif');
+			canvas.style.zIndex = '1';
```

Comment:
Slightly weird to see zindex specified as a string but it's only a personal preference, no need to change it